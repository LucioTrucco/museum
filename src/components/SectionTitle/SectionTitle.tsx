import React, { memo } from 'react';
import { AppText } from '../AppText';
import { colors } from '@/styles';
import { StyleSheet } from 'react-native';

interface SectionTitleProps {
  title: string;
}
export const component = ({ title }: SectionTitleProps) => {
  return <AppText.Headline2 style={styles.title}>{title}</AppText.Headline2>;
};

const styles = StyleSheet.create({
  title: {
    paddingHorizontal: 10,
    marginVertical: 20,
    fontWeight: 'bold',
    color: colors.primary,
  },
});

export const SectionTitle = memo(component);
