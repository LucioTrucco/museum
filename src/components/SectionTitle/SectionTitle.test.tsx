import React from 'react';
import { render } from '@testing-library/react-native';
import { SectionTitle } from './SectionTitle';

describe('tests => SectionTitle', () => {
  let component: any;

  beforeEach(() => {
    component = render(<SectionTitle title="This is a title value" />);
  });

  it('should render SectionTitle', () => {
    component.getByText('This is a title value');
  });

  it('should match snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
