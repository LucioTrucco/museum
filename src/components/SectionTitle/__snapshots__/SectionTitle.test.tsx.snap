// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests => SectionTitle should match snapshot 1`] = `
<Text
  style={
    [
      {
        "fontSize": 54,
        "fontWeight": "600",
        "letterSpacing": 0.25,
      },
      {
        "color": "#b50938",
        "fontWeight": "bold",
        "marginVertical": 20,
        "paddingHorizontal": 10,
      },
    ]
  }
>
  This is a title value
</Text>
`;
